### ISSUE TITLE
Not working in 3.26

### ISSUE THREAD

**Issue by bluuub** [schuhumi/gnome-shell-extension-onboard-integration#7]
Hi there,
I used this extension on a Surface Pro 3 running Arch Linux + GNOME. Used to work perfectly but the update to 3.26 broke functionality (it can not be enabled anymore). Any chance you will update this? Or maybe you can point me in the right direction where to start investigating the issue (I am no GNOME developer).

Thanks!

**Comment by bluuub** [comment]
I just realized that it can still be enabled. The Onboard part seems to work just fine. The problem is that caribou opens on every touch, which is very annoying since the tablet can basically not be used like that. So the disabling of caribou seems to be not available in G3.26.

**Comment by rickysarraf** [comment]
Yes. I realized the same recently. BTW, this should now be reported to Onboard upstream. Can you please file a bug here: https://launchpad.net/onboard

**Comment by rickysarraf** [comment]
The interface in 3.26 has changed.

```
Oct 23 09:39:10 priyasi gnome-shell[15016]: JS ERROR: TypeError: Main.keyboard.Show is not a function
                                            Onboard<.ShowAnyKeyboard@/usr/share/gnome-shell/extensions/Onboard_Indicator@onboard.org/extension.js:193:9
                                            wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                            enable_show_gesture/<@/usr/share/gnome-shell/extensions/Onboard_Indicator@onboard.org/extension.js:225:17
                                            EdgeDragAction<.vfunc_gesture_end@resource:///org/gnome/shell/ui/edgeDragAction.js:82:13
                                            wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
```

**Comment by yesrod** [comment]
New interface is at https://github.com/GNOME/gnome-shell/blob/master/js/ui/keyboard.js , specifically the KeyboardManager() function.

TL;DR:
- ```Show()``` is now ```open()```
- ```Hide()``` is now ```close()```
- ```_keyboardVisible``` is now ```visible```
- Don't need to manually call ```_sync()``` when changing config/variables/whatever that does, the constructor calls ```_syncEnabled()``` which appears to make sync "just work"

The changes I'm working on are for https://extensions.gnome.org/extension/1061/on-screen-keyboard-button/ but they should apply here as well.  Not sure what the "right" way is to handle these changes across multiple versions of GNOME Shell - is there a variable somewhere that provides the current version of GNOME Shell?

**Comment by PeterPorker3** [comment]
Well I with I saw this thread before I spent 4 hours figuring all that out... but anyways today I have forked the repo and made some changes to make it work with more recent versions of GNOME Shell (3.36 in this case). You can find the source here: https://github.com/PeterPorker3/gnome-shell-extension-onboard-integration and I am waiting for it to get approved on extensions.gnome.org. Feel free to contribute there, it's very far from perfect and any help is appreciated!
